llvm/lib/Transforms/Scalar/InferAddressSpaces.cpp

```diff
@@ -335,13 +349,25 @@ static bool isAddressExpression(const Value &V, const DataLayout &DL,
   }
   case Instruction::IntToPtr:
     return isNoopPtrIntCastPair(Op, DL, TTI);
+  case Instruction::Load: {
+    const auto *LI = cast<LoadInst>(Op);
+    if (LI->getType()->isPtrOrPtrVectorTy()) {
+      // Heuristic: treat load-of-GV as an address expression only if the GV is
+      // locally accessed by load and store.
+      if (const auto *GV = dyn_cast<GlobalVariable>(LI->getPointerOperand()))
+        return isLocallyAccessedBySimpleLoadsStores(GV, LI->getFunction());
+    }
+    return TTI->getAssumedAddrSpace(&V) != UninitializedAddressSpace;
```

Comment:
The assumed address space check should take precedence over this special treatment

Reply:
> The assumed address space check should take precedence over this special treatment

this code is deleted